From 4b1628fc24cfcf4145bacb1a7ac5f363c7013126 Mon Sep 17 00:00:00 2001
From: Jianhong Chen <chenjh@rock-chips.com>
Date: Wed, 17 Jun 2015 14:02:46 +0800
Subject: [PATCH] rk3368-tb_8846: dts: modify battery node

	1) add dc det pinctrl
	2) disable usb adapter
	3) set input charge current 3000mA

Signed-off-by: Jianhong Chen <chenjh@rock-chips.com>
---
 arch/arm64/boot/dts/rk3368-tb_8846.dts | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/rk3368-tb_8846.dts b/arch/arm64/boot/dts/rk3368-tb_8846.dts
index 3a3d4a07188f..082692810945 100644
--- a/arch/arm64/boot/dts/rk3368-tb_8846.dts
+++ b/arch/arm64/boot/dts/rk3368-tb_8846.dts
@@ -244,7 +244,9 @@
 			gpio0_c2: gpio0-c2 {
 				rockchip,pins = <0 GPIO_C2 RK_FUNC_GPIO &pcfg_pull_down>;
 			};
-
+                        gpio0_c1: gpio0-c1 {
+                                rockchip,pins = <0 GPIO_C1 RK_FUNC_GPIO &pcfg_pull_up>;
+                        };
 			//to add
 		};
 
@@ -878,7 +880,8 @@
 &rk818 {
 	gpios =<&gpio0 GPIO_A1 GPIO_ACTIVE_HIGH>,<&gpio0 GPIO_A0 GPIO_ACTIVE_LOW>;
 	rk818,system-power-controller;
-
+        pinctrl-names = "default";
+        pinctrl-0 = <&gpio0_c1>;
 rk818,support_dc_chg = <1>;/*1:dc chg; 0:usb chg*/
 	regulators {
 
@@ -1050,15 +1053,17 @@ rk818,support_dc_chg = <1>;/*1:dc chg; 0:usb chg*/
                 design_capacity = <8650>;
                 design_qmax = <8800>;
                 max_overcharge = <100>;
-                max_input_currentmA  = <2000>;
+                max_input_currentmA  = <3000>;
                 max_chrg_currentmA = <1800>;
                 max_charge_voltagemV = <4200>;
                 sleep_enter_current = <600>;
                 sleep_exit_current = <600>;
                 power_off_thresd = <3400>;
                 chrg_diff_voltagemV = <0>;
-                support_usb_adp = <1>;
+                virtual_power = <1>;
+                support_usb_adp = <0>;
                 support_dc_adp = <1>;
+                dc_det_gpio = <&gpio0 GPIO_C1 GPIO_ACTIVE_LOW>;
         };
 };
 
-- 
2.35.3

